commitlintplugins.test.ts

```diff
@@ -4,12 +4,77 @@ function runCommitLintOnMsg(inputMsg: string) {
     return spawnSync('npx', ['commitlint', '--verbose'], { input: inputMsg });
 }
 
+test('body-prose1', () => {
+    let commitMsgWithLowercaseBodyStart =
+        "foo: this is only a title" + "\n\n" + "bla blah bla.";
+    let bodyProse1 = runCommitLintOnMsg(commitMsgWithLowercaseBodyStart);
+    expect(bodyProse1.status).not.toBe(0);
+});
+
+
+test('body-prose2', () => {
+    let commitMsgWithNumbercaseBodyStart =
+        "foo: this is only a title" + "\n\n" + "1234 bla blah bla.";
+    let bodyProse2 = runCommitLintOnMsg(commitMsgWithNumbercaseBodyStart);
+    expect(bodyProse2.status).toBe(0);
+});
+
+
+test('body-prose3', () => {
+    let commitMsgWithUrl =
+        "foo: this is only a title" + "\n\n" + "someUrl://blahblah.com";
+    let bodyProse3 = runCommitLintOnMsg(commitMsgWithUrl);
+
+    // because URLs can bypass the rule
+    expect(bodyProse3.status).toBe(0);
+});
+
+
+test('body-prose4', () => {
+    let commitMsgWithFootnoteUrl =
+        "foo: this is only a title" + "\n\n" + "Bla blah[1] bla.\n\n[1] someUrl://blahblah.com";
+    let bodyProse4 = runCommitLintOnMsg(commitMsgWithFootnoteUrl);
+
+    // because URLs in footer can bypass the rule
+    expect(bodyProse4.status).toBe(0);
+});
+
+
+test('body-prose5', () => {
+    let commitMsgWithBugUrl =
+        "foo: this is only a title" + "\n\n" + "Fixes someUrl://blahblah.com";
+    let bodyProse5 = runCommitLintOnMsg(commitMsgWithBugUrl);
+
+    // because URLs in "Fixes <URL>" sentence can bypass the rule
+    expect(bodyProse5.status).toBe(0);
+});
+
+
+test('body-prose6', () => {
+    let commitMsgWithBlock =
+        "foo: this is only a title" + "\n\n" + "Bar baz.\n\n```\n" + 'if (foo) {bar(); }' + "\n```";
```

Comment:
why are you wrapping the code block with single-quotes, and why are you splitting this string so many times? IMO it's simpler to just write:

```
let commitMsgWithBlock =
        "foo: this is only a title\n\nBar baz.\n\n```\nif (foo) { bar(); }\n```";
```

Reply:
Done